app.js

```diff
@@ -0,0 +1,299 @@
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+uninstall \
+${chartName} \
+${installationParameters.join(" ")} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+async function runCommand(pluginParameters) {
+  const {
+    kubeCertificate,
+    cliCommand,
+  } = pluginParameters;
+
+  let result;
+
+  await helpers.temporaryFileSentinel(
+    [kubeCertificate],
+    async (certificateFilePath) => {
+      const [certificatePath, certificateFileName] = splitDirectory(certificateFilePath);
+      const certificateVolumeDefinition = docker.createVolumeDefinition(certificatePath);
+      const volumeDefinitions = [
+        certificateVolumeDefinition,
+      ];
+
+      const authenticationParameters = generateAuthenticationParameters({
+        certificateFilePath: `$${certificateVolumeDefinition.mountPoint.name}/${certificateFileName}`,
+        kubeToken: pluginParameters.kubeToken,
+        kubeApiServer: pluginParameters.kubeApiServer,
+        kubeUser: pluginParameters.kubeUser,
+      });
+
+      const dockerEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.mountPoint.name]: cur.mountPoint.value,
+        }),
+        {},
+      );
+
+      const shellEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+${sanitizeCommand(cliCommand)} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+function generateAuthenticationParameters(pluginParams) {
+  const params = [];
+
+  params.push("--kube-ca-file", pluginParams.certificateFilePath);
+  params.push("--kube-token", pluginParams.kubeToken);
+  params.push("--kube-apiserver", pluginParams.kubeApiServer);
+  params.push("--kube-as-user", pluginParams.kubeUser);
```

Comment:
Why not define these elements directly in the `params` array?
```js
const params = [
  "--kube-ca-file", pluginParams.certificateFilePath,
  "--kube-token", pluginParams.kubeToken,
  "--kube-apiserver", pluginParams.kubeApiServer,
  "--kube-as-user", pluginParams.kubeUser,
];
```
Or you can even directly return an array. It seems like this function is just a wrapper for mapping object fields into an array, is it really necessary? I can see it is used in `temporaryFileSentinel` callback functions, but once you merge these functions into one and extract it, `generateAuthenticationParameters` function will not be needed anymore.

Reply:
Fixed